### DISCUSSION TITLE
Quasar prop naming is inconsistent.

### DISCUSSION THREAD

**Discussion by natankeddem** [zauberzeug/nicegui#1744] [General]
Anyone else find props naming inconsistent? 3 different elements with essentially the same prop functionality: `bordered`, `outline`, `outlined`? Is there a reason Quasar does this? Maybe I am just not smart enough to understand?🤷‍♂️

**Comment by falkoschindler** [comment]
Interesting. I didn't notice...

Can you name examples, i.e. which Quasar elements have props `bordered`, `outline` or `outlined`? Maybe their reference page indicates an explanation.

Apart from that I assume a mix of
- historical reasons (maybe they started with an adjective like "outlined" and switched to the CSS name "outline" or vice versa), and
- technical differences behind the scenes (e.g. see [What is the difference between outline and border CSS properties?](https://stackoverflow.com/questions/1158515/what-is-the-difference-between-outline-and-border-css-properties)).

**Comment by natankeddem** [comment]
The 3 elements are:
- input https://quasar.dev/vue-components/input which uses `outlined`
- button https://quasar.dev/vue-components/button/ which uses `outline`
- card https://quasar.dev/vue-components/card/ which uses `bordered`

It makes it so I constantly need to look at the docs to make sure I am picking the right one.

**Reply by falkoschindler** [reply]
Oh yes, that's annoying! 🙄

**Reply by natankeddem** [reply]
Glad I could make someone else annoyed with this today! Working with these I couldn't believe no one was bringing it up on their github repo. I am sure it is historic, but I wish they would have picked a name and ran with it I'll even accept the `outline` which actually makes a `border`...

**Comment by natankeddem** [comment]
Out of curiosity I looked at button, the prop is `outline` but it uses a border:
https://github.com/zauberzeug/nicegui/blob/bfa1f6eb30dda4b22a5f9b9c41ae931731874896/nicegui/static/quasar.css#L469-L471C2

**Reply by falkoschindler** [reply]
🤦‍♂️
